clang/test/Sema/bool-compare.c

```diff
@@ -85,7 +85,7 @@ void f(int x, int y, int z) {
   if ((a<y) != -1) {}// expected-warning {{comparison of constant -1 with boolean expression is always true}}
 
   if ((a<y) == z) {} // no warning
-  if (a>y<z)      {} // expected-error {{comparisons like 'X<=Y<=Z' don't have their mathematical meaning}}
+  if (a>y<z)      {} // expected-error {{comparisons like 'a > y < z' don't have their mathematical meaning}}
```

Comment:
Hmm.... @AaronBallman will probably have to comment here, but I'm not sure this is better?  The point of `comparisons like` here was to simplify the diagnostics to something that is `like` what is there, but not actually.

This patch ends up giving us really awful warning/diagnostics in the case where these are more than simple variables, as its just dumping the entire expression.  So I don't know if I like this patch.

Reply:
Yes i think the word `like` should be removed here, but I am thinking should we prefer the previous diagnostics or the one with actual expression

Reply:
I don't like the one with the actual expression in it, as that is going to get arbitrarily large.  That said, I DO wonder if there is value in using the actual operators instead of just `<=` for both sides (so not printing the expression, just the operators, and still `X`, `Y`, `Z`).  That would require keeping `like` though.

Reply:
Yes, I got your point also it will not be great with large expressions :D, I am thinking should we change the message to be different or just the old one with actual operators 🤔

Reply:
I feel like making it so it prints the actual operators is simple enough to where we can just do that. I agree that we probably don’t want to print the entire expression because that can get rather messy.

Reply:
Further thought: I haven't seen the 'change' you're proposing to the message, but am willing to keep an open mind.  But I also think just changing to the actual operators is a distinct improvement.